### PR TITLE
enable managedDNS for GCP

### PR DESCRIPTION
**Pull request by staebler**

### CODE REVIEW

**Comment by staebler** [comment]
I need to use managed DNS for the e2e gcp tests, too. I can do that as part of this PR, or I can do it as a follow-up PR.

**Comment by staebler** [comment]
> I need to use managed DNS for the e2e gcp tests, too. I can do that as part of this PR, or I can do it as a follow-up PR.

I have added it as part of this PR.

File: contrib/pkg/adm/managedns/enable.go [hunk 1]
```diff
@@ -117,23 +120,39 @@ func (o *Options) Run(dynClient client.Client, args []string) error {
...
+		hc.Spec.ExternalDNS = &hivev1.ExternalDNSConfig{
+			GCP: &hivev1.ExternalDNSGCPConfig{
+				Credentials: corev1.LocalObjectReference{Name: manageDNSCredentialsSecret},
+			},
+		}
+	default:
+		log.WithField("cloud", o.Cloud).Fatal("unsupported cloud")
 	}
 
+	log.Infof("created Route53 credentials secret: %s", credsSecret.Name)
```
Comment (dgoodwin) [inline contrib/pkg/adm/managedns/enable.go]: -Route53

**Comment by dgoodwin** [comment]
Could you also update the using-hive.md docs as this mentions not being available for GCP and is currently Route53 specific.

**Pull request merged**